analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/DoNotHardcodeCredentialsBase.cs

```diff
@@ -168,8 +168,11 @@ private void CheckAppSettings(SonarAnalysisContext context, CompilationAnalysisC
 
         private string IssueMessage(string variableName, string variableValue)
         {
-            var bannedWords = FindCredentialWords(variableName, variableValue);
-            if (bannedWords.Any())
+            if (string.IsNullOrWhiteSpace(variableValue))
+            {
+                return null;
+            }
+            else if (FindCredentialWords(variableName, variableValue) is var bannedWords && bannedWords.Any())
```

Comment:
Could be merged.

Reply:
> if (!string.IsNullOrWhiteSpace(variableValue) && FindCredentialWords(variableName, variableValue) is var bannedWords && bannedWords.Any())

No, I need to stop also for the 2nd elseif in that case. It doesn't make sense to explore the regex against empty